File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        ans = count0 = j = 0
        for i in range(len(nums)):
            if nums[i] == 0:
                count0 += 1
            while j < i and count0 > 1:
                if nums[j] == 0:
                    count0 -= 1
                j += 1
            ans = max(ans, i - j)
        return ans

File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestSubarray = function(nums) {
  let [j, count0, ans] = [0, 0, 0];
  
  for (let i = 0; i < nums.length; i++) {
    if (!nums[i]) count0++;
    while (count0 > 1) {
      if (!nums[j++]) count0--;
    }  
    
    ans = Math.max(ans, i - j);
  }
  
  return ans;
};